omniNotes/src/androidTest/java/it/feio/android/omninotes/utils/SortNotes.java

```diff
@@ -0,0 +1,65 @@
+package it.feio.android.omninotes.utils;
+
+import android.widget.ListView;
+
+import androidx.test.espresso.ViewInteraction;
+
+import java.util.List;
+
+import it.feio.android.omninotes.R;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static androidx.test.espresso.matcher.ViewMatchers.isChecked;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withParent;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static it.feio.android.omninotes.utils.EspressoUtils.checkIfMatches;
+import static it.feio.android.omninotes.utils.EspressoUtils.childAtPosition;
+import static it.feio.android.omninotes.utils.EspressoUtils.clickOnButton;
+import static it.feio.android.omninotes.utils.EspressoUtils.getText;
+import static it.feio.android.omninotes.utils.NewNoteStringVariables.titleButton;
+import static org.hamcrest.Matchers.allOf;
+
+public class SortNotes {
+    /**
+     * This method is sorting the list of notes and stores the titles in an array and return the array with the titles
+     *
+     * @param arrayWithTitles - in this parameter are stored the titles after sorting by title
+     * @param numberOfTimes   - this is the parameter of how many notes were added
+     * @return -  the array with titles
+     */
+    public static List<String> getArrayWithTitlesAfterSort(List<String> arrayWithTitles, int numberOfTimes) {
+        int index = 0;
+
+        sortBy(titleButton);
+        while (index < numberOfTimes) {
+            arrayWithTitles.add(getText(allOf(withParent(childAtPosition(childAtPosition(childAtPosition(withId(R.id.list), index), 0), 1)), withId(R.id.note_title))));
+            index++;
+        }
+        return arrayWithTitles;
+    }
+
+    /**
+     * This method sorts the list with notes by a text
+     *
+     * @param buttonText - the sorting is done by the text selected e.g. Title, Creation date
+     */
+    public static void sortBy(String buttonText) {
+        ViewInteraction sortButton = onView(withId(R.id.menu_sort));
+        clickOnButton(sortButton);
+        do {
+            clickOnButton(onView(withText(buttonText)));
+            clickOnButton(sortButton);
+
+        } while (!isSelected());
+        clickOnButton(onView(withText(buttonText)));
+    }
+
+    /**
+     * This method returns true if the button is checked
+     */
+    private static boolean isSelected() {
+        return checkIfMatches(onView(allOf(withId(R.id.radio), withParent(allOf(withId(R.id.content), withParent(childAtPosition(isAssignableFrom(ListView.class), 1)))))), isChecked());
```

Comment:
you use one particular index in this check (index 1), but you call this method above in sortBy method, which is called for multiple buttons (title and creation date), which are at different indexes

Reply:
yes, you are right, I added a hashmap which contains the titles of the sorting options...and based on the selected option, will get the index and than check if this particular option is or isn't selected